File: position.py
def input_values():
    N = input("Введите число N: ")
    P = int(input("Введите позицию P: "))
    return N, P

def output_value(N, P):
    if P <= 0 or P > len(N):
        print("Некорректная позиция P")
    else:
        result = N[-P]
        print(result)


N, P = input_values()
output_value(N, P)

File: position.php
<?php
// Функция для ввода числа и позиции
function input(&$N, &$P) {
    echo "Введите число N: ";
    $N = trim(fgets(STDIN)); // fgets(STDIN) считывает строку, введённую пользователем, из стандартного ввода
    echo "Введите позицию P: ";
    $P = intval(trim(fgets(STDIN))); // intval() преобразует введённое значение в целое число
}

// Функция для вывода символа на указанной позиции
function output($N, $P) {
    if ($P <= 0 || $P > strlen($N)) {
        echo "Некорректная позиция P" . PHP_EOL; // PHP_EOL используется для переноса строки
    } else {
        $result = $N[strlen($N) - $P];
        echo $result . PHP_EOL;
    }
}

$N = "";
$P = 0;

input($N, $P);
output($N, $P);
?>